academy/4-my-own-chain/cosmjs-gui.md

```diff
@@ -293,34 +293,34 @@ constructor(props: IMenuContainerProps) {
    }
     ```
 
-    Note that it creates it only once by saving the client in the state.
+    Note that this creates it only once by saving the client in the state.
 
-Instantiating a `CheckersSigningStargateClient` is a bit more involved as you will see later on.
+Instantiating a `CheckersSigningStargateClient` is more involved, as you will see later on.
 
 First, make use of what you already prepared. Why choose to start with `MenuContainer`? Because that is where all the games are listed.
 
 ## Show all games
 
-To show all games, you only need the read-only `StargateClient`. That is convenient as there is no need to worry about private keys, for now. Where do you query for the games?
+To show all games, you only need the read-only `StargateClient`. There is no need to worry about private keys for now. Where do you query for the games?
 
 ### All games container
 
-Look into `src/components/Menu/MenuContainer.tsx`'s `componentDidMount` method. It is accessing the browser storage for saved games. You can replace the storage logic with:
+Look into `src/components/Menu/MenuContainer.tsx` for the `componentDidMount` method, which accesses the browser storage for saved games. You can replace the storage logic with the following steps:
 
 * Obtain a `StargateClient` with the `getStargateClient` method.
-* Obtain the blockchain games. Without pagination to keep it simple.
+* Obtain the blockchain games (without pagination).
 * Convert the blockchain games to the format the component expects.
 * Save them in `saved: IGameInfo[]` of the component's state.
 
-Your first thought may be to add a new function to `CheckersStargateClient` to handle the call and the conversion, or to add a helper function that takes a client as a parameter.
+A good next step is to add a new function to `CheckersStargateClient` to handle the call and the conversion, or to add a helper function that takes a client as a parameter.
 
 ### A specific GUI method
 
-However, in the future you may want to reuse the `CheckersStargateClient` code in another GUI or for backend scripts. So, to avoid polluting the code, and to avoid a helper where you need to pass a Stargate client as parameter, you can add [an extension method](https://www.c-sharpcorner.com/article/learn-about-extension-methods-in-typescript/#:~:text=Extension%2Dmethod%20gives%20you%20the,any%20data%2Dtype%20you%20want) to `CheckersStargateClient`.
+In the future you may want to reuse the `CheckersStargateClient` code in another GUI or for backend scripts. To avoid polluting the code, and to avoid a helper where you need to pass a Stargate client as parameter, add [an extension method](https://www.c-sharpcorner.com/article/learn-about-extension-methods-in-typescript/#:~:text=Extension%2Dmethod%20gives%20you%20the,any%20data%2Dtype%20you%20want) to `CheckersStargateClient`.
```

Comment:
```suggestion
In the future you may want to reuse the `CheckersStargateClient` code in another GUI or for backend scripts. To avoid polluting the code, and to avoid a less-elegant helper where you need to pass a Stargate client as parameter, add [an extension method](https://www.c-sharpcorner.com/article/learn-about-extension-methods-in-typescript/#:~:text=Extension%2Dmethod%20gives%20you%20the,any%20data%2Dtype%20you%20want) to `CheckersStargateClient`.
```